vulnerabilities/importers/alpine_linux.py

```diff
@@ -20,124 +20,192 @@
+    assert anchor_tags, f"No anchor tags found in {advisory_directory_link}"
+    for anchor_tag in anchor_tags:
+        if anchor_tag.text.endswith("json"):
+            yield f"{advisory_directory_link}{anchor_tag.text}"
+
+
+def process_record(record: dict) -> List[AdvisoryData]:
+    """
+    Return a list of AdvisoryData objects by processing data
+    present in that `record`
+    """
+    if not record["packages"]:
+        raise Exception(f'"packages" not found in this {record!r}')
+
+    advisories: List[AdvisoryData] = []
+
+    for package in record["packages"]:
+        assert package["pkg"]
+        attributes = ["distroversion", "reponame", "archs"]
+        for attribute in attributes:
+            assert attribute in record
+        loaded_advisories = load_advisories(
+            package["pkg"],
+            record["distroversion"],
+            record["reponame"],
+            record["archs"],
+        )
+        advisories.extend(loaded_advisories)
+    return advisories
+
+
+def load_advisories(
+    pkg_infos: Mapping[str, Any],
+    distroversion: str,
+    reponame: str,
+    archs: List[str],
+) -> Iterable[AdvisoryData]:
+    """
+    Yields AdvisoryData by mapping data from `pkg_infos`
+    and form PURL for AffectedPackages by using
+    `distroversion`, `reponame`, `archs`
+    """
+    assert pkg_infos.get("name"), '"name" is not available in package'
+
+    for version, fixed_vulns in pkg_infos["secfixes"].items():
+
+        assert fixed_vulns, f"No fixed vulnerabilities in {version}"
+
+        for vuln_ids in fixed_vulns:
+            assert isinstance(vuln_ids, str)
+            vuln_ids = vuln_ids.split()
+            aliases = []
+            vuln_id = vuln_ids[0]
+            # check for valid vuln ID, if there is valid vuln ID then iterate over
+            # the remaining elements of the list else iterate over the whole list
+            # and also check if the initial element is a reference or not
+            if is_cve(vuln_id):
+                aliases = [vuln_id]
+                vuln_ids = vuln_ids[1:]
+            references = []
+            for reference_id in vuln_ids:
+
+                if reference_id.startswith("XSA"):
+                    references.append(XsaReference.from_id(xsa_id=reference_id))
+
+                elif reference_id.startswith("ZBX"):
+                    references.append(ZbxReference.from_id(zbx_id=reference_id))
+
+                elif reference_id.startswith("wnpa-sec"):
+                    references.append(WireSharkReference.from_id(wnpa_sec_id=reference_id))
+
+            qualifiers = {
+                "distroversion": distroversion,
+                "reponame": reponame,
+            }
+
+            affected_packages = []
+
+            try:
+                fixed_version = AlpineLinuxVersion(version)
+            except Exception as e:
+                raise Exception(f"{version!r} Not a valid Alpine Version") from e
+            assert isinstance(archs, List)
```

Comment:
This is an unnecessary assert

Reply:
This is to ensure that the `archs` is a List only, and not any other dataType, imagine a case if archs is a str, and we would be iterating over every character of string rather than iterating over list of arch